tests/compile-fail/never_say_never.rs

```diff
@@ -4,7 +4,7 @@
 fn main() {
     let y = &5;
     let x: ! = unsafe {
-        *(y as *const _ as *const !) //~ ERROR entered unreachable code
+        *(y as *const _ as *const !) //~ ERROR tried to access a dead local variable
```

Comment:
Okay actually what happens is that the variable is just never marked live. We require all variables to be marked live before usage. rustc seemingly never marks variables of type `!` live (which makes sense), leading to this error.